projects/plugins/jetpack/extensions/blocks/videopress/deprecated/v2/utils.js

```diff
@@ -25,31 +24,29 @@ export const ASPECT_RATIOS = [
  */
 export const filterVideoPressClasses = ( className, videoPressClassNames ) => {
 	// Filter out VideoPress specific styles from className attribute.
-	const aspectRatioClassNames = ASPECT_RATIOS.reduce(
-		( accumulator, { className: ratioClass } ) => {
-			accumulator[ ratioClass ] = false;
-			return accumulator;
-		},
-		{ 'wp-has-aspect-ratio': false }
-	);
 
-	const customClasses = classnames( className, {
+	const aspectRatioClassNames = ASPECT_RATIOS.reduce( ( accumulator, ratio ) => {
+		accumulator.push( ratio.className );
+		return accumulator;
+	}, [] );
+
+	const customClassTokenList = new TokenList( className );
+	customClassTokenList.remove(
 		...aspectRatioClassNames,
-		'wp-block-embed': false,
-		'is-type-video': false,
-		'is-provider-videopress': false,
-	} );
+		'wp-block-embed',
+		'is-type-video',
+		'is-provider-videopress'
+	);
 
-	// Filter the custom CSS classes from the VideoPress specific class list.
-	const customClassList = new TokenList( customClasses );
-	let videoPressClasses = classnames( className, videoPressClassNames );
+	const videoPressClassTokenList = new TokenList( className );
+	videoPressClassTokenList.add( videoPressClassNames );
 
-	customClassList.forEach( customClass => {
-		videoPressClasses = videoPressClasses.replace( customClass, '' );
+	customClassTokenList.forEach( customClass => {
+		videoPressClassTokenList.remove( customClass );
 	} );
```

Comment:
Looks like you could simplify this too, `...` works on a TokenList.
```suggestion
	videoPressClassTokenList.remove( ...customClassTokenList );
```

Reply:
Nice. I did experiment with passing an array of items and passing the class string, but your version works splendidly.